src/viur/core/prototypes/tree.py

```diff
@@ -286,6 +286,20 @@ def pathToKey(self, key: db.Key):
 
     ## External exposed functions
 
+    @exposed
+    def index(self, skelType: SkelType = "node", parententry: t.Optional[db.Key | int | str] = None, **kwargs):
+        if not parententry:
+            repos = self.getAvailableRootNodes(**kwargs)
+            match len(repos):
+                case 0:
+                    raise errors.Unauthorized()
+                case 1:
+                    parententry = repos.pop()["key"]
```

Comment:
I wouldn't pop it without a copy of it. If `getAvailableRootNodes` provides anything cached this would be modified.